src/utils/createDispatcher.js

```diff
@@ -0,0 +1,38 @@
+import getContextValue from './getContextValue';
+
+const createDispatcher = (context) => ({
+  readContext: (suppliedContext) => {
+    return getContextValue(context, suppliedContext);
+  },
+  useContext: (suppliedContext) => {
+    return getContextValue(context, suppliedContext);
```

Comment:
Can you explain why this is needed?
Both `react-shallow-renderer` and `react-reconciler` just return `suppliedContext._currentValue` from what I can see.

Reply:
I believe we support more context stuff than they do. SuppliedContext is normally just a new object of the context-type, so it will have inital values only. We want to use a context set by a Context.Provider higher up in the component-tree which is what `getContextValue` finds.
I'm not sure if this is a good way of doing it, but this system was already implemented for `Context.Consumer`. I just made it work with `useContext`